### PR TITLE
feat: Automatically dispatch reward pool into markets in recurring tr…

### PR DESCRIPTION
**Pull request by ze97286**
…ansfers

Closes #5318 

```json
{
     "data-node": "553"
}
```

### CODE REVIEW

**Review by jeremyletang — CHANGES_REQUESTED** [review]
Sorry requesting changes to block the merge just now as I don't have the time to go through it now and would like to read the PR before it get merged.

**Review by EVODelavega — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**
